data-prepper-plugins/prometheus-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/prometheus/service/PrometheusSinkService.java

```diff
@@ -4,49 +4,415 @@
+                byte[] bytes = null;
+                if (event.getMetadata().getEventType().equals("METRIC")) {
+                    Remote.WriteRequest message = null;
+                    if (event instanceof JacksonGauge) {
+                        JacksonGauge jacksonGauge = (JacksonGauge) event;
+                        message = buildRemoteWriteRequest(jacksonGauge.getTime(),
+                                jacksonGauge.getStartTime(), jacksonGauge.getValue(), jacksonGauge.getAttributes(),jacksonGauge.getName());
+                    } else if (event instanceof JacksonSum) {
+                        JacksonSum jacksonSum = (JacksonSum) event;
+                        message = buildRemoteWriteRequest(jacksonSum.getTime(),
+                                jacksonSum.getStartTime(), jacksonSum.getValue(), jacksonSum.getAttributes(), jacksonSum.getName());
+                    } else if (event instanceof JacksonSummary) {
+                        JacksonSummary jacksonSummary = (JacksonSummary) event;
+                        message = buildRemoteWriteRequest(jacksonSummary.getTime(),
+                                jacksonSummary.getStartTime(), jacksonSummary.getSum(), jacksonSummary.getAttributes(), jacksonSummary.getName());
+                    } else if (event instanceof JacksonHistogram) {
+                        JacksonHistogram jacksonHistogram = (JacksonHistogram) event;
+                        message = buildRemoteWriteRequest(jacksonHistogram.getTime(),
+                                jacksonHistogram.getStartTime(), jacksonHistogram.getSum(), jacksonHistogram.getAttributes(), jacksonHistogram.getName());
+                    } else if (event instanceof JacksonExponentialHistogram) {
+                        JacksonExponentialHistogram jacksonExpHistogram = (JacksonExponentialHistogram) event;
+                        message = buildRemoteWriteRequest(jacksonExpHistogram.getTime(),
+                                jacksonExpHistogram.getStartTime(), jacksonExpHistogram.getSum(), jacksonExpHistogram.getAttributes(), jacksonExpHistogram.getName());
+                    } else {
+                        LOG.error("No valid Event type found");
+                    }
+                    bytes = message.toByteArray();
+                }
+                if (event.getEventHandle() != null) {
+                    this.bufferedEventHandles.add(event.getEventHandle());
+                }
+                HttpEndPointResponse failedHttpEndPointResponses = null;
+                try {
+                    failedHttpEndPointResponses = pushToEndPoint(bytes);
+                } catch (IOException e) {
+                    LOG.info("Error while pushing to the end point");
+                }
+                if (failedHttpEndPointResponses != null) {
+                    logFailedData(failedHttpEndPointResponses);
+                    releaseEventHandles(Boolean.FALSE);
+                } else {
```

Comment:
This is not true when exception occurs at line 166.

Reply:
@omkarmmore95  and @mallikagogoi7  Could you please have a look and clarify?

Reply:
@dinujoh The code has been  updated accordingly